test/framework/src/main/java/org/opensearch/test/InternalTestCluster.java

```diff
@@ -514,6 +516,10 @@ public Collection<Class<? extends Plugin>> getPlugins() {
         return plugins;
     }
 
+    public Map<Class<? extends Plugin>, Class<? extends Plugin>> getExtendedPlugins() {
```

Comment:
This is not needed, the `PluginInfo` is fully sufficient to reconstruct these relations

Reply:
Sorry, I was traveling last week and couldn't reply earlier. Should I consider changing [NodeConfigurationSource](https://github.com/opensearch-project/OpenSearch/blob/main/test/framework/src/main/java/org/opensearch/test/NodeConfigurationSource.java#L62-L65)?

Reply:
I think we should adapt `NodeConfigurationSource` to match new classpathPlugins shape that Node accepts.